ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/InternalCustomerCenter.kt

```diff
@@ -70,6 +76,10 @@ internal fun InternalCustomerCenter(
         }
     }
 
+    LaunchedEffect(Unit) {
+        viewModel.trackImpression()
+    }
```

Comment:
Not sure if this is the best way to do it. Let me know if there's another way

Reply:
This works most times, but could potentially be called multiple times if the activity is recreated (like in config changes like changing device rotation for example). So what we've done is to cache the data in the view model, since the view model survives activity recreations, and then you can call this method like `trackImpresionIfNeeded`, which will only track if the VM doesn't already have cached data.

I suggest looking at how this looks for paywalls: https://github.com/RevenueCat/purchases-android/blob/a07303d64091c856ee511f013fbe3f4dc280f0ea/ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt#L177-L182